### ISSUE TITLE
Is it possible to separate the collection window from the preview window?

### ISSUE THREAD

**Issue by wmailn** [BelledonneCommunications/linphone-android#1880]
Hi,

	In my use, I found that when resetting the preview window, the camera would stop and then reopen, which resulted in a bit of a lag in the preview window.
	May I ask if it is possible to set different Windows for preview and collection respectively, so that when switching the preview window, there is no need to restart camera collection.
	Thank you

1. **Describe the bug** (mandatory)
	The camera is turned on and off repeatedly.

2. **To Reproduce** (mandatory)
	Repeatedly destroy and reconfigure the preview window to appear.

3. **Expected behavior** (mandatory)
	I hope the camera doesn't have to start working again when I switch the preview window.

4. **Please complete the following information** (mandatory)
	- Device: [Private device]
	- OS: [Android 13]
	- Version of the App [NULL,Immediate use linphone-sdk]
	- Version of the SDK [5.2.30]
	- Where you did got it from (local build)
	- My Android is The original Android

5. **SDK logs** (mandatory)
	--xiaomi--lc--[M:setVideoInfo]: setVideoInfo  width:352, height:288, fps:15, bitrate:512000, orientation:0
	--xiaomi--lc--[M:getVideoDevicesList]: ...
	--xiaomi--lcj--: getVideoDevicesList i:0 , id:BackFacingCamera0
	--xiaomi--lcj--: getVideoDevicesList i:1 , id:FrontFacingCamera1
	--xiaomi--lc--[M:setVideoDevice]: start .... id:BackFacingCamera0
	--xiaomi--lc--[M:setVideoDevice]:find WebCam:  id:BackFacingCamera0, name:Android video name
	--xiaomi--lc---[M:notifyEventListener]: event:[10001] start ...
	--xiaomi--lc--[M:setAmrParam]: bit:12200, payloadFormat:1, ptime:20
	--xiaomi--lc--[M:setNetinfo]: type:1, ip:40.40.0.11, audio:13526, video:13532
	--xiaomi--lc--[M:setLocalNetinfo]: ip:40.40.0.11, audio:13526, video:13532
	--xiaomi--lc--[M:setNetinfo]: type:2, ip:182.168.90.11, audio:13502, video:13508
	--xiaomi--lc--[M:setRemoteNetinfo]: ip:182.168.90.11, audio:13502, video:13508
	--xiaomi--lc--[M:initAudio]: start... captureType:[2],  isGroupCall:[NO]
	Fail to set rtp port reusable: Protocol not available.
	RtpSession bound to [0.0.0.0] ports [13526] [54987]
	Fail to increase socket's recv buffer size with SO_RCVBUFFORCE: Operation not permitted.
	Fail to increase socket's send buffer size with SO_SNDBUFFORCE: Operation not permitted.
	Fail to increase socket's recv buffer size with SO_RCVBUFFORCE: Operation not permitted.
	--xiaomi--lc--[M:initAudio]: audio_stream_new locport:[13526]
	cannot set noise gate mode to [0] because no volume send
	RtpSession [0xae80f400] sending to rtp 182.168.90.11:13502 rtcp 182.168.90.11:13502 
	Stun packet of length 20 sent on rtp for session [0xae80f400] 
	Stun packet of length 20 sent on rtcp for session [0xae80f400] 
	msandroid_sound_read_new
	msandroid_sound_write_new
	Hardware sample rate is 48000
	audio_stream_start_from_io: create encoder, decoder and resamplers.
	speex_lib_ctl init with neon ? 1
	speex_lib_ctl init with neon ? 1
	cannot set echo limiter to mode [0] because no volume send
	cannot set noise gate mode to [0] because no volume send
	set_nchannels 1
	set_nchannels 1
	MKVRecorder: initialisation
	speex_lib_ctl init with neon ? 1
	No such filter with id 115
	Configuring av recorder with audio format type=audio;encoding=pcmu;rate=8000;channels=1;fmtp=''
	MKVRecorder: set pin #1 format. type=audio;encoding=pcmu;rate=8000;channels=1;fmtp=''
	target bitrate not set for stream [0xae85ef80] using payload's bitrate is 12200
	Setting audio encoder network bitrate to [12200] on stream [0xae85ef80]
	Setting maxbitrate=-3800 to AMR-NB encoder.
	Could not set maxbitrate 12200 to AMR-NB encoder.
	RtpSession: target upload bandwidth set to 12200
	No information available for [XiaoMi/13/msm8974],
	Android < 8, using default sound device descriptor
	Sound device information for [XiaoMi/13/msm8974] is: builtin=[no], delay=[250] ms
	No information available for [XiaoMi/13/msm8974],
	Android < 8, using default sound device descriptor
	Sound device information for [XiaoMi/13/msm8974] is: builtin=[no], delay=[250] ms
	Could not apply gain on sent RTP packets: gain control wasn't activated. Use audio_stream_enable_gain_control() before starting the stream.
	No information available for [XiaoMi/13/msm8974],
	Android < 8, using default sound device descriptor
	Sound device information for [XiaoMi/13/msm8974] is: builtin=[no], delay=[250] ms
	Could not apply gain on received RTP packet: gain control wasn't activated. Use audio_stream_enable_gain_control() before starting the stream.
	get_nchannels 0
	Filter MSAmrEnc does not implement the MS_FILTER_GET_NCHANNELS method
	Filter MSAmrEnc does not implement the MS_FILTER_GET_SAMPLE_RATE method
	configuring MSAndSoundRead:0xae883778-->MSAmrEnc:0xae883430 from rate [8000] to rate [8000] and from channel [1] to channel [1]
	get_nchannels 0
	Filter MSAmrDec does not implement the MS_FILTER_GET_NCHANNELS method
	Filter MSAmrDec does not implement the MS_FILTER_GET_SAMPLE_RATE method
	configuring MSAmrDec:0xae883200-->MSAndSoundWrite:0xae883938 from rate [8000] to rate [48000] and from channel [1] to channel [1]
	ms_filter_link: MSAndSoundRead:0xae883778,0-->MSResample:0xae8832e0,0
	ms_filter_link: MSResample:0xae8832e0,0-->MSAudioMixer:0xae883890,0
	ms_filter_link: MSAudioMixer:0xae883890,0-->MSAmrEnc:0xae883430,0
	ms_filter_link: MSAmrEnc:0xae883430,0-->MSRtpSend:0xae883900,0
	ms_filter_link: MSRtpRecv:0xae883708,0-->MSAmrDec:0xae883200,0
	ms_filter_link: MSAmrDec:0xae883200,0-->MSTee:0xae883548,0
	ms_filter_link: MSTee:0xae883548,0-->MSResample:0xae8833c0,0
	ms_filter_link: MSResample:0xae8833c0,0-->MSAndSoundWrite:0xae883938,0
	ms_filter_link: MSAudioMixer:0xae8834a0,1-->MSResample:0xae883660,0
	ms_filter_link: MSResample:0xae883660,0-->MSUlawEnc:0xae883698,0
	ms_filter_link: MSUlawEnc:0xae883698,0-->MSMKVRecorder:0xae883580,1
	ms_filter_link: MSItcSource:0xae8835b8,0-->MSMKVRecorder:0xae883580,0
	ms_filter_link: MSAudioMixer:0xae883890,1-->MSAudioMixer:0xae8834a0,0
	ms_filter_link: MSTee:0xae883548,1-->MSAudioMixer:0xae8834a0,1
	ms_filter_link: MSAudioMixer:0xae8834a0,0-->MSFileRec:0xae883468,0
	andsnd_read_preprocess
	andsnd_read_preprocess
	Configuring recorder with [16] bits  rate [8000] nchanels [1] buff size [1280], chunk size [160]
	--xiaomi--lc--[M:sound_read_setup]: capture_type:7
	Priority used: 99
	MSAudio MSTicker priority increased to maximum.
	ms_ticker_set_time_func: ticker's time method updated.
	Initializing speex resampler in mode [voip] 
	Initializing speex resampler in mode [voip] 
	andsnd_write_preprocess
	--xiaomi--aso--[M:msandroid_sound_write_preprocess]:
	Configuring player with [16] bits  rate [48000] nchanels [1] buff size [7680] chunk size [1920]
	Initializing speex resampler in mode [voip] 
	Filter MSRtpRecv is already being scheduled; nothing to do.
	--xiaomi--lc--[M:initAudio]: locport:13526, mDstUrl:182.168.90.11, mRemAudioPort:13502, isGroupCall:NO
	--xiaomi--lc--[M:initAudio]: end
	--xiaomi--lc--[M:initVideo]: start... isGroupCall:[NO]
	--xiaomi--msc--: get_feature key:FEATURE_LINPHONE_ENCODE_V_FORMAT, def:H264
	--xiaomi--as--[M:initVideo] FEATURE_LINPHONE_ENCODE_V_FORMAT:H264
	Fail to set rtp port reusable: Protocol not available.
	RtpSession bound to [0.0.0.0] ports [13532] [45088]
	Fail to increase socket's recv buffer size with SO_RCVBUFFORCE: Operation not permitted.
	Fail to increase socket's send buffer size with SO_SNDBUFFORCE: Operation not permitted.
	Fail to increase socket's recv buffer size with SO_RCVBUFFORCE: Operation not permitted.
	No information available for [XiaoMi/13/msm8974],
	Android < 8, using default sound device descriptor
	Sound device information for [XiaoMi/13/msm8974] is: builtin=[no], delay=[250] ms
	--xiaomi--lc--[M:initVideo]:  video_stream_new  ...
	Setting preview video size 352x288
	Setting video size 352x288 on stream [0xae87d840]
	--xiaomi--lc--[M:initVideo]: video_stream_set_video_size  width:352, height:288
	--xiaomi--lc--[M:initVideo]: video_stream_set_fps :15
	[Legacy Capture] Instanciating Android VIDEO capture MS filter
	[Legacy Capture] Creating AndroidReaderContext for Android VIDEO capture filter
	[Legacy Capture] Init of Android VIDEO capture filter (0xa2866600)
	[Legacy Capture] getHelperClassGlobalRef (env: 0xae882280)
	[Legacy Capture] Android SDK version found is 22
	RtpSession [0xae815600] sending to rtp 182.168.90.11:40002 rtcp 182.168.90.11:40002 
	Using permissive algorithm
	Stun packet of length 20 sent on rtp for session [0xae815600] 
	Stun packet of length 20 sent on rtcp for session [0xae815600] 
	Stun packet of length 20 sent on rtp for session [0xae80f400] 
	Stun packet of length 20 sent on rtcp for session [0xae80f400] 
	MSAudioMixer [0xae883890] is entering bypass mode.
	--xiaomi--lc--[M:getVideoDevice]: start 
	--xiaomi--lc--[M:getVideoDevice]: end getVideoDevice id:(null)
	--xiaomi--lc--[M:getVideoDevice]: start 
	--xiaomi--lc--[M:getVideoDevice]: end getVideoDevice id:(null)
	Codec video/avc successfully created.
	H26xNalPacker: max payload size set to 1240 bytes
	No information available for [XiaoMi/13/msm8974],
	Android < 8, using default sound device descriptor
	Sound device information for [XiaoMi/13/msm8974] is: builtin=[no], delay=[250] ms
	No such filter with id 78
	Using the default video configuration list
	Max target bitrate not set for stream [0xae87d840], but using payload type's bitrate [0]
	Limiting bitrate of video encoder to 512000 bits/s for stream [0xae87d840]
	H26xEncoder: video configuration set (bitrate=128000b/s, fps=15.000000, vsize=352x288)
	H26xEncoder: video configuration now:  bitrate=512000b/s, fps=15.000000, vsize=352x288
	RtpSession: target upload bandwidth set to 512000
	H26xEncoder: AVPF enabled
	video_set_device_rotation : 0
	selectNearestResolutionAvailable: 0, 352x288
	mediastreamer16 supported resolutions :
	mediastreamer	1280x960
	mediastreamer	1280x720
	mediastreamer	864x480
	mediastreamer	720x576
	mediastreamer	800x480
	mediastreamer	720x480
	mediastreamer	768x432
	mediastreamer	640x480
	mediastreamer	576x432
	mediastreamer	480x320
	mediastreamer	384x288
	mediastreamer	352x288
	mediastreamer	320x240
	mediastreamer	240x160
	mediastreamer	176x144
	mediastreamer	160x120
	mediastreamerresolution selection done (352, 288, 0)
	[Legacy Capture] Camera selected resolution is: 352x288 (requested: 352x288) with downscaling?0
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	[Legacy Capture] Swapped resolution width and height to : 288x352
	Camera video preview size changed on videostream [0xae87d840]
	Output video size adjusted to match camera resolution (288x352)
	Setting sent vsize=288x352, fps=15.000000
	H26xEncoder: video configuration set (bitrate=512000b/s, fps=15.000000, vsize=352x288)
	H26xEncoder: video configuration now:  bitrate=512000b/s, fps=15.000000, vsize=288x352
	ms_filter_link: MSAndroidVideoCapture:0xae883af8,0-->MSPixConv:0xae883d60,0
	ms_filter_link: MSPixConv:0xae883d60,0-->MSTee:0xae883d28,0
	ms_filter_link: MSTee:0xae883d28,0-->MSMediaCodecH264Encoder:0xae883b30,0
	ms_filter_link: MSMediaCodecH264Encoder:0xae883b30,0-->MSRtpSend:0xae883a50,0
	Codec video/avc successfully created.
	MediaCodecDecoder: starting decoder with following parameters:
	ity=0, max-width=1920, color-format=2135033992}
	MediaCodecDecoder: decoder successfully started. In-force parameters:
	ity=0, max-width=1920, color-format=2135033992}
	MediaCodecH264Decoder: got device info: { 'XiaoMi', '13', 'msm8974' }
	No such filter with id 78
	H26xDecoder: enabling AVPF mode
	H26xDecoder: freeze on error disabled
	msasync.c: worker thread has no tasks.
	Video stream[0xae87d840] thumbnail[0] direction[0]: set display mode 2 to filter MSAndroidTextureDisplay
	ms_filter_link: MSRtpRecv:0xae883fc8,0-->MSTee:0xae883a88,0
	ms_filter_link: MSTee:0xae883a88,1-->MSItcSink:0xae883ac0,0
	configure_itc(): format set to type=video;encoding=H264;vsize=0x0;fps=0.000000;fmtp=''
	ms_filter_link: MSTee:0xae883a88,0-->MSMediaCodecH264Decoder:0xae883e78,0
	ms_filter_link: MSMediaCodecH264Decoder:0xae883e78,0-->MSAndroidTextureDisplay:0xae883f58,0
	ms_filter_link: MSTee:0xae883d28,1-->MSAndroidTextureDisplay:0xae883f58,1
	[Legacy Capture] Preprocessing of Android VIDEO capture filter
	MSVideo MSTicker priority left to normal.
	[Legacy Capture] Starting Android camera '0' (rotation:0)
	startRecording(0, 352, 288, 30, 0, 2726716928)
	First estimation
	Stun packet of length 20 sent on rtcp for session [0xae80f400] 
	FOCUS_MODE_CONTINUOUS_VIDEO is supported, let's use it
	Searching for closest fps range from 30000
	The closest fps range is w=7500,h=30000
	Camera preview orientation: 90
	Returning camera object: android.hardware.Camera@31f4a199
	[Legacy Capture] Preprocessing of Android VIDEO capture filter done
	MediaCodecEncoder: configuring MediaCodec with the following parameters:
	3992, profile=1, height=352, width=288, bitrate=460800, priority=0, i-frame-interval=2, level=512, bitrate-mode=1}
	MediaCodecEncoder: encoder successfully configured. In-force parameters:

	MediaCodecEncoder: encoder successfully started
	Filter MSRtpRecv is already being scheduled; nothing to do.
	--xiaomi--lc--[M:initVideo]: locport:13532, const_remip:182.168.90.11, mRemVideoPort:40002, isGroupCall:NO
	--xiaomi--lc--[M:initVideo]: end
	--xiaomi--lc--[M:startVideoReceive]: ...
	--xiaomi--atd---[M:android_texture_display_show_video]: show_video:TRUE
	--xiaomi--ms--[M:media_stream_enable_rtp_recv]: val:TRUE
	B-TrunC Call Analyzer: rtp start receiver
	--xiaomi--lc--[M:startVideoSend]: ...
	H26xEncoder: VFU requested
	--xiaomi--ms--[M:media_stream_enable_rtp_send]: val:TRUE
	B-TrunC Call Analyzer: rtp start send
	--xiaomi--lc--[M:startAudioSend]: ...
	--xiaomi--ms--[M:media_stream_enable_rtp_send]: val:TRUE
	B-TrunC Call Analyzer: rtp start send
	--xiaomi--lc--[M:startVideoSend]: ...
	H26xEncoder: VFU requested
	--xiaomi--ms--[M:media_stream_enable_rtp_send]: val:TRUE
	B-TrunC Call Analyzer: rtp start send
	--xiaomi--lc--[M:startAudioReceive]: ...
	--xiaomi--ms--[M:media_stream_enable_rtp_recv]: val:TRUE
	B-TrunC Call Analyzer: rtp start receiver
	--xiaomi--lc--[M:startVideoReceive]: ...
	--xiaomi--atd---[M:android_texture_display_show_video]: show_video:TRUE
	--xiaomi--ms--[M:media_stream_enable_rtp_recv]: val:TRUE
	B-TrunC Call Analyzer: rtp start receiver
	--xiaomi--lc--[M:startAudioSend]: ...
	--xiaomi--ms--[M:media_stream_enable_rtp_send]: val:TRUE
	B-TrunC Call Analyzer: rtp start send
	--xiaomi--lc--[M:startVideoSend]: ...
	H26xEncoder: VFU requested
	--xiaomi--ms--[M:media_stream_enable_rtp_send]: val:TRUE
	B-TrunC Call Analyzer: rtp start send
	--xiaomi--lc--[M:startAudioReceive]: ...
	--xiaomi--ms--[M:media_stream_enable_rtp_recv]: val:TRUE
	B-TrunC Call Analyzer: rtp start receiver
	--xiaomi--lc--[M:startVideoReceive]: ...
	--xiaomi--atd---[M:android_texture_display_show_video]: show_video:TRUE
	--xiaomi--ms--[M:media_stream_enable_rtp_recv]: val:TRUE
	B-TrunC Call Analyzer: rtp start receiver
	[Linphonexiaomi] ..............setVideoPreviewViewApi....................
	[Linphonexiaomi] Preview window surface is directly available for texture view [com.xiaomi.ui.call.GotaTextureView{12e5aa57 V.ED..C. ........ 507,240-705,482 #7f080459 app:id/view_gotacall_status_gl2jniview_localevideo}]
	--xiaomi--lc---[M:setNativePreviewWindowId]: window:[0xa2fb5a00] start ...
	--xiaomi--lc--: setNativePreviewWindowId: video_stream_started ...
	[Legacy Capture] Preview capture window set for the 1st time (win: 0x200912 rotation:0)
	mediastreamersetPreviewDisplaySurface(android.hardware.Camera@31f4a199, com.xiaomi.ui.call.GotaTextureView{12e5aa57 V.ED..C. ........ 507,240-705,482 #7f080459 app:id/view_gotacall_status_gl2jniview_localevideo})
	Camera [android.hardware.Camera@31f4a199] preview started
	Camera video preview size changed on videostream [0xae87d840]
	--xiaomi--lc---[M:setNativePreviewWindowId]: end ...
	Stun packet of length 20 sent on rtp for session [0xae815600] 
	[Linphonexiaomi] Rendering window surface is directly available for texture view [com.xiaomi.ui.call.GotaTextureView{1bfa3944 V.ED..C. ........ 0,0-720,1280 #7f08045a app:id/view_gotacall_status_gl2jniview_remotevideo}]
	--xiaomi--lc---[M:setNativeVideoWindowId]: window:[0xa2fb5a00] start ...
	--xiaomi--lc--: setNativeVideoWindowId: video_stream_started ...
	Stun packet of length 20 sent on rtcp for session [0xae815600] 
	[TextureView Display][Filter=0xae883f58] New windowId jobject ptr is 0x100916, current one is 0x0
	[TextureView Display][Filter=0xae883f58] Took global ref on 0x100916, windowId is now 0x10091a
	--xiaomi--lc---[M:setNativeVideoWindowId]: end ...
	First estimation
	Switching rtp destination from 182.168.90.11:40002 to 182.168.90.11:13508 for session [0xae815600]
	[TextureView Display][Filter=0xae883f58] Initializing context for windowId 0x10091a
	[TextureView Display] NativePreviewWindowId 0x10091a is a TextureView, let's get it's SurfaceTexture first
	[TextureView Display] Creating Surface from SurfaceTexture 0x10000d
	[TextureView Display] Surface created 0x10091e for SurfaceTexture 0x10000d
	[TextureView Display][Filter=0xae883f58] Chosen format for windowId 0x10091a is 4
	[TextureView Display][Filter=0xae883f58] Surface size for windowId 0x10091a is 720x1280
	[ogl_display] ogl_display_new : 0xa2beb300
	[ogl_functions] Function : Fail to load plugin libGLEW.so: dlopen failed: library "libGLEW.so" not found
	[ogl_display] init opengles_display (720 x 1280, gl initialized:0)
	OpenGL version string: OpenGL ES 3.0 V@84.0 AU@05.01.00.042.030 (CL@)
	OpenGL extensions: GL_AMD_compressed_ATC_texture GL_AMD_performance_monitor GL_AMD_program_binary_Z400 GL_EXT_debug_label GL_EXT_debug_marker GL_EXT_discard_framebuffer GL_EXT_robustness GL_EXT_texture_format_BGRA8888 GL_EXT_texture_type_2_10_10_10_REV GL_NV_fence GL_OES_compressed_ETC1_RGB8_texture GL_OES_depth_texture GL_OES_depth24 GL_OES_EGL_image GL_OES_EGL_sync GL_OES_EGL_image_external GL_OES_element_index_uint GL_OES_fbo_render_mipmap GL_OES_fragment_precision_high GL_OES_get_program_binary GL_OES_packed_depth_stencil GL_OES_depth_texture_cube_map GL_OES_rgb8_rgba8 GL_OES_standard_derivatives GL_OES_texture_3D GL_OES_texture_float GL_OES_texture_half_float GL_OES_texture_half_float_linear GL_OES_texture_npot GL_OES_vertex_half_float GL_OES_vertex_type_10_10_10_2 GL_OES_vertex_array_object GL_QCOM_alpha_test GL_QCOM_binning_control GL_QCOM_driver_control GL_QCOM_perfmon_global_mode GL_QCOM_extended_get GL_QCOM_extended_get2 GL_QCOM_tiled_rendering GL_QCOM_writeonly_rendering GL_EXT_sRGB GL_EXT_sRGB_write_control GL_EXT_texture_sRGB_decode GL_EXT_texture_filter_anisotropic GL_EXT_multisampled_render_to_texture GL_EXT_color_buffer_float GL_EXT_color_buffer_half_float GL_EXT_disjoint_timer_query 
	OpenGL vendor: Qualcomm
	OpenGL renderer: Adreno (TM) 330
	OpenGL version: OpenGL ES 3.0 V@84.0 AU@05.01.00.042.030 (CL@)
	OpenGL GLSL version: OpenGL ES GLSL ES 3.00
	[ogl_display] resize opengles_display (720 x 1280, gl initialized:0)
	[ogl_display] Falling back to legacy shaders.
	[ogl_display] OpenGL program info: 
	[TextureView Display][Filter=0xae883f58] Context initialized for windowId 0x10091a
	--xiaomi--aso--[M:msandroid_write_cb] ...
	B-TrunC Call Analyzer audio : AudioTrack write first data
	Stun packet of length 20 sent on rtcp for session [0xae80f400] 
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	H26xEncoder: requesting I-frame to the encoder.
	AMediaCodec_setParams() request-sync key transfered to Bundle with value 0
	MediaCodecEncoder: I-frame requested to MediaCodec
	Stun packet of length 20 sent on rtcp for session [0xae815600] 
	MSVideo MSTicker: We are late of 85 miliseconds.
	MediaCodecEncoder: output format has changed.
	MediaCodecEncoder: new output format:
	n=0,limit=8,capacity=8], what=1869968451, mime=video/avc, csd-0=java.nio.ByteArrayBuffer[position=0,limit=17,capacity=17]}
	H264ParameterSetsInserter: I-frame detected
	H264ParameterSetsInserter: parameter sets inserted
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	MediaCodecDecoder: new paramter sets received
	MediaCodecDecoder: key frame received
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	MediaCodecDecoder: output buffers has changed
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	MediaCodecDecoder: output format has changed
	MediaCodecDecoder: new format 288x352 :
	ity=80], mime=video/raw, crop-top=0, crop-right=287, slice-height=352, color-format=2141391876, height=352, width=288, what=1869968451, crop-bottom=351, crop-left=0, stride=384}
	H26xDecoder: first frame decoded 288x352
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	[ogl_display] allocate_gl_textures: allocated new textures[0] (512 x 512)
	Switching rtcp destination from 182.168.90.11:40002 to 182.168.90.11:13508 for session [0xae815600]
	video processing delay = { min=0ms, mean=0.0ms, max=0ms } (RtpSession=0xae815600)
	[Linphonexiaomi] Rendering surface texture [android.graphics.SurfaceTexture@23bb9c5e] destroyed
	[Linphonexiaomi] Current rendering surface texture is no longer available
	[Linphonexiaomi] Releasing window surface texture [android.graphics.SurfaceTexture@23bb9c5e]
	--xiaomi--lc---[M:setNativeVideoWindowId]: window:[0x0] start ...
	--xiaomi--lc--: setNativeVideoWindowId: video_stream_started ...
	[TextureView Display][Filter=0xae883f58] New windowId jobject ptr is 0x0, current one is 0x10091a
	--xiaomi--lc---[M:setNativeVideoWindowId]: end ...
	[Linphonexiaomi] Preview surface texture [android.graphics.SurfaceTexture@2f3a683f] destroyed
	[Linphonexiaomi] Current preview surface texture is no longer available
	[Linphonexiaomi] Releasing preview window surface texture [android.graphics.SurfaceTexture@2f3a683f]
	--xiaomi--lc---[M:setNativePreviewWindowId]: window:[0x0] start ...
	--xiaomi--lc--: setNativePreviewWindowId: video_stream_started ...
	[Legacy Capture] Preview capture window changed (oldwin: 0x200912 newwin: 0x0 rotation:0)
	stopRecording(android.hardware.Camera@31f4a199)
	Sending RTCP SR compound message on session [0xae80f400].
	[TextureView Display][Filter=0xae883f58] eglSwapBuffers failure for windowId 0x0: 0
	[TextureView Display][Filter=0xae883f58] Destroying context for windowId 0x0
	[ogl_display] uninit opengles_display (gl initialized:1)
	[TextureView Display][Filter=0xae883f58] OGL display destroyed
	[TextureView Display][Filter=0xae883f58] EGL display destroyed for windowId 0x0
	[TextureView Display][Filter=0xae883f58] Window released for windowId 0x0
	[TextureView Display][Filter=0xae883f58] Surface released for windowId 0x0
	[TextureView Display][Filter=0xae883f58] Surface destroyed for windowId 0x0
	[TextureView Display][Filter=0xae883f58] Context destroyed for windowId 0x0
	startRecording(0, 352, 288, 30, 0, 2726716928)
	FOCUS_MODE_CONTINUOUS_VIDEO is supported, let's use it
	Searching for closest fps range from 30000
	The closest fps range is w=7500,h=30000
	Camera preview orientation: 90
	Returning camera object: android.hardware.Camera@b64be55
	Camera video preview size changed on videostream [0xae87d840]
	--xiaomi--lc---[M:setNativePreviewWindowId]: end ...
	H264ParameterSetsInserter: I-frame detected
	H264ParameterSetsInserter: parameter sets inserted
	MediaCodecDecoder: no buffer available for queuing this frame ! Decoder is too slow.
	Request sending of PLI on videostream [0xae87d840]
	Sending RTCP SR compound message on session [0xae815600].
	jitter buffer stable: target-size: 59.999999 ms, effective-size: 13.880597 (min: 60 nom: 60, max: 500)
	jitter buffer rls stats: count=249, clockrate=8000, offset=-227.964 clock_ratio=0.997851, capped_offset=-1833466026 capped_clock_ratio=0.997851, max_ts_deviation=7.000000 prev_max_ts_deviation=480.000000, deviation=7, RLS VARIABLES: P[0][0]=0.000000, P[1][0]=-0.000000, P[0][1]=-0.000000, P[1][1]=0.011812
	[Linphonexiaomi] ..............setVideoPreviewViewApi....................
	[Capture TextureView] Rotating preview texture by 0
	[Capture TextureView] Rotating preview texture by 0
	[Linphonexiaomi] Preview window surface texture [android.graphics.SurfaceTexture@16a1206a] is available for texture view [com.xiaomi.ui.call.GotaTextureView{75bc50c V.ED.... ........ 0,0-200,244}]
	--xiaomi--lc---[M:setNativePreviewWindowId]: window:[0xa2fb5a50] start ...
	--xiaomi--lc--: setNativePreviewWindowId: video_stream_started ...
	[Legacy Capture] Preview capture window set for the 1st time (win: 0x77e rotation:0)
	mediastreamersetPreviewDisplaySurface(android.hardware.Camera@b64be55, com.xiaomi.ui.call.GotaTextureView{75bc50c V.ED.... ........ 0,0-200,244})
	Camera [android.hardware.Camera@b64be55] preview started
	Camera video preview size changed on videostream [0xae87d840]
	--xiaomi--lc---[M:setNativePreviewWindowId]: end ...
	jitter buffer stable: target-size: 269.588888 ms, effective-size: 15.731707 (min: 0 nom: 0, max: 500)
	jitter buffer rls stats: count=307, clockrate=90000, offset=2339.01 clock_ratio=1.0101, capped_offset=-1956533865 capped_clock_ratio=1.010104, max_ts_deviation=24263.000000 prev_max_ts_deviation=21865.000000, deviation=3407, RLS VARIABLES: P[0][0]=0.000000, P[1][0]=-0.000000, P[0][1]=-0.000000, P[1][1]=0.012043
	 H26x decoder: FPS: 12.221214
	MSVideo MSTicker: We are late of 361 miliseconds.
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	Captured mean fps=2.694730, expected=15.000000
	Sending RTCP RR compound message on session [0xae815600].
	[Linphonexiaomi] ..............setVideoPreviewViewApi....................
	[Linphonexiaomi] Found an existing preview TextureView, let's destroy it first
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	[Linphonexiaomi] Rendering window surface texture [android.graphics.SurfaceTexture@3677495b] is available for texture view [com.xiaomi.ui.call.GotaTextureView{1bfa3944 V.ED..C. ........ 0,0-720,1280 #7f08045a app:id/view_gotacall_status_gl2jniview_remotevideo}]
	--xiaomi--lc---[M:setNativeVideoWindowId]: window:[0xa2fb5a70] start ...
	--xiaomi--lc--: setNativeVideoWindowId: video_stream_started ...
	[TextureView Display][Filter=0xae883f58] New windowId jobject ptr is 0x10084e, current one is 0x0
	[TextureView Display][Filter=0xae883f58] Took global ref on 0x10084e, windowId is now 0x10079e
	--xiaomi--lc---[M:setNativeVideoWindowId]: end ...
	[Linphonexiaomi] Preview window surface texture [android.graphics.SurfaceTexture@208625f8] is available for texture view [com.xiaomi.ui.call.GotaTextureView{12e5aa57 V.ED..C. ........ 507,240-705,482 #7f080459 app:id/view_gotacall_status_gl2jniview_localevideo}]
	--xiaomi--lc---[M:setNativePreviewWindowId]: window:[0xa2fb5a50] start ...
	--xiaomi--lc--: setNativePreviewWindowId: video_stream_started ...
	[Legacy Capture] Preview capture window changed (oldwin: 0x77e newwin: 0x10077e rotation:0)
	stopRecording(android.hardware.Camera@b64be55)
	[Capture TextureView] Rotating preview texture by 0
	[Capture TextureView] Rotating preview texture by 0
	[TextureView Display][Filter=0xae883f58] Initializing context for windowId 0x10079e
	[TextureView Display] NativePreviewWindowId 0x10079e is a TextureView, let's get it's SurfaceTexture first
	[TextureView Display] Creating Surface from SurfaceTexture 0x100025
	[TextureView Display] Surface created 0x100906 for SurfaceTexture 0x100025
	[TextureView Display][Filter=0xae883f58] Chosen format for windowId 0x10079e is 4
	[TextureView Display][Filter=0xae883f58] Surface size for windowId 0x10079e is 720x1280
	[ogl_display] ogl_display_new : 0xa2beb900
	[ogl_functions] Function : Fail to load plugin libGLEW.so: dlopen failed: library "libGLEW.so" not found
	[ogl_display] init opengles_display (720 x 1280, gl initialized:0)
	[ogl_display] resize opengles_display (720 x 1280, gl initialized:0)
	[ogl_display] Falling back to legacy shaders.
	[ogl_display] OpenGL program info: 
	[TextureView Display][Filter=0xae883f58] Context initialized for windowId 0x10079e
	startRecording(0, 352, 288, 30, 0, 2726716928)
	Sending RTCP SR compound message on session [0xae80f400].
	FOCUS_MODE_CONTINUOUS_VIDEO is supported, let's use it
	Searching for closest fps range from 30000
	The closest fps range is w=7500,h=30000
	Camera preview orientation: 90
	Returning camera object: android.hardware.Camera@e7fead1
	mediastreamersetPreviewDisplaySurface(android.hardware.Camera@e7fead1, com.xiaomi.ui.call.GotaTextureView{12e5aa57 V.ED..C. ........ 507,240-705,482 #7f080459 app:id/view_gotacall_status_gl2jniview_localevideo})
	Camera [android.hardware.Camera@e7fead1] preview started
	Camera video preview size changed on videostream [0xae87d840]
	--xiaomi--lc---[M:setNativePreviewWindowId]: end ...
	MediaCodecDecoder: no buffer available for queuing this frame ! Decoder is too slow.
	MediaCodecDecoder: no buffer available for queuing this frame ! Decoder is too slow.
	Request sending of PLI on videostream [0xae87d840]
	[ogl_display] allocate_gl_textures: allocated new textures[0] (512 x 512)
	Sending RTCP SR compound message on session [0xae815600].
	MSVideo MSTicker: We are late of 423 miliseconds.
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	H264ParameterSetsInserter: I-frame detected
	H264ParameterSetsInserter: parameter sets inserted
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	[Linphonexiaomi] Rendering surface texture [android.graphics.SurfaceTexture@3677495b] destroyed
	[Linphonexiaomi] Current rendering surface texture is no longer available
	--xiaomi--lc---[M:setNativeVideoWindowId]: window:[0x0] start ...
	--xiaomi--lc--: setNativeVideoWindowId: video_stream_started ...
	[TextureView Display][Filter=0xae883f58] New windowId jobject ptr is 0x0, current one is 0x10079e
	--xiaomi--lc---[M:setNativeVideoWindowId]: end ...
	[Linphonexiaomi] Releasing window surface texture [android.graphics.SurfaceTexture@3677495b]
	[Linphonexiaomi] Preview surface texture [android.graphics.SurfaceTexture@208625f8] destroyed
	[Linphonexiaomi] Current preview surface texture is no longer available
	--xiaomi--lc---[M:setNativePreviewWindowId]: window:[0x0] start ...
	--xiaomi--lc--: setNativePreviewWindowId: video_stream_started ...
	[Legacy Capture] Preview capture window changed (oldwin: 0x10077e newwin: 0x0 rotation:0)
	stopRecording(android.hardware.Camera@e7fead1)
	[Linphonexiaomi] Releasing preview window surface texture [android.graphics.SurfaceTexture@208625f8]
	[Capture TextureView] Rotating preview texture by 0
	[Capture TextureView] Rotating preview texture by 0
	[TextureView Display][Filter=0xae883f58] eglSwapBuffers failure for windowId 0x0: 0
	[TextureView Display][Filter=0xae883f58] Destroying context for windowId 0x0
	[ogl_display] uninit opengles_display (gl initialized:1)
	[TextureView Display][Filter=0xae883f58] OGL display destroyed
	[TextureView Display][Filter=0xae883f58] EGL display destroyed for windowId 0x0
	[TextureView Display][Filter=0xae883f58] Window released for windowId 0x0
	[TextureView Display][Filter=0xae883f58] Surface released for windowId 0x0
	[TextureView Display][Filter=0xae883f58] Surface destroyed for windowId 0x0
	[TextureView Display][Filter=0xae883f58] Context destroyed for windowId 0x0
	startRecording(0, 352, 288, 30, 0, 2726716928)
	[Capture TextureView] Rotating preview texture by 0
	[Capture TextureView] Rotating preview texture by 0
	Excess of audio samples in capture side bytes=320 (20 ms)
	FOCUS_MODE_CONTINUOUS_VIDEO is supported, let's use it
	Searching for closest fps range from 30000
	The closest fps range is w=7500,h=30000
	Camera preview orientation: 90
	Returning camera object: android.hardware.Camera@33d72037
	Camera video preview size changed on videostream [0xae87d840]
	--xiaomi--lc---[M:setNativePreviewWindowId]: end ...
	[Linphonexiaomi] ..............setVideoPreviewViewApi....................
	[Linphonexiaomi] Preview window surface is directly available for texture view [com.xiaomi.ui.call.GotaTextureView{14a8d936 V.ED.... ........ 0,0-200,244}]
	--xiaomi--lc---[M:setNativePreviewWindowId]: window:[0xa2fb5a00] start ...
	--xiaomi--lc--: setNativePreviewWindowId: video_stream_started ...
	[Legacy Capture] Preview capture window set for the 1st time (win: 0x1006a2 rotation:0)
	mediastreamersetPreviewDisplaySurface(android.hardware.Camera@33d72037, com.xiaomi.ui.call.GotaTextureView{14a8d936 V.ED.... ........ 0,0-200,244})
	MediaCodecDecoder: no buffer available for queuing this frame ! Decoder is too slow.
	Request sending of PLI on videostream [0xae87d840]
	Camera [android.hardware.Camera@33d72037] preview started
	Camera video preview size changed on videostream [0xae87d840]
	--xiaomi--lc---[M:setNativePreviewWindowId]: end ...
	Sending RTCP SR compound message on session [0xae815600].
	Sending RTCP SR compound message on session [0xae80f400].
	video processing delay = { min=0ms, mean=0.0ms, max=0ms } (RtpSession=0xae815600)
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	Sending RTCP RR compound message on session [0xae815600].
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	H264ParameterSetsInserter: I-frame detected
	H264ParameterSetsInserter: parameter sets inserted
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	[Capture TextureView] Rotating preview texture by 0
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	[Capture TextureView] Rotating preview texture by 0
	[Capture TextureView] Rotating preview texture by 0
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	[Capture TextureView] Rotating preview texture by 0
	jitter buffer stable: target-size: 59.999999 ms, effective-size: 13.673470 (min: 60 nom: 60, max: 500)
	jitter buffer rls stats: count=499, clockrate=8000, offset=-254.965 clock_ratio=0.999386, capped_offset=-1833466053 capped_clock_ratio=0.999386, max_ts_deviation=305.000000 prev_max_ts_deviation=243.000000, deviation=97, RLS VARIABLES: P[0][0]=0.000000, P[1][0]=-0.000000, P[0][1]=-0.000000, P[1][1]=0.007269
	[Capture TextureView] Rotating preview texture by 0
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	[Capture TextureView] Rotating preview texture by 0
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	[Capture TextureView] Rotating preview texture by 0
	[Linphonexiaomi] ..............setVideoPreviewViewApi....................
	[Linphonexiaomi] Found an existing preview TextureView, let's destroy it first
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	MSVideo MSTicker: We are late of 425 miliseconds.
	[Capture TextureView] Rotating preview texture by 0
	[Capture TextureView] Rotating preview texture by 0
	[Linphonexiaomi] Rendering window surface texture [android.graphics.SurfaceTexture@3ff3a9a4] is available for texture view [com.xiaomi.ui.call.GotaTextureView{1bfa3944 V.ED..C. ........ 0,0-720,1280 #7f08045a app:id/view_gotacall_status_gl2jniview_remotevideo}]
	--xiaomi--lc---[M:setNativeVideoWindowId]: window:[0xa2fb5a70] start ...
	--xiaomi--lc--: setNativeVideoWindowId: video_stream_started ...
	[TextureView Display][Filter=0xae883f58] New windowId jobject ptr is 0x100946, current one is 0x0
	[TextureView Display][Filter=0xae883f58] Took global ref on 0x100946, windowId is now 0x10094a
	--xiaomi--lc---[M:setNativeVideoWindowId]: end ...
	[TextureView Display][Filter=0xae883f58] Initializing context for windowId 0x10094a
	[TextureView Display] NativePreviewWindowId 0x10094a is a TextureView, let's get it's SurfaceTexture first
	[TextureView Display] Creating Surface from SurfaceTexture 0x10003d
	[TextureView Display] Surface created 0x10094e for SurfaceTexture 0x10003d
	[TextureView Display][Filter=0xae883f58] Chosen format for windowId 0x10094a is 4
	[TextureView Display][Filter=0xae883f58] Surface size for windowId 0x10094a is 720x1280
	[Linphonexiaomi] Preview window surface texture [android.graphics.SurfaceTexture@1147630d] is available for texture view [com.xiaomi.ui.call.GotaTextureView{12e5aa57 V.ED..C. ........ 507,240-705,482 #7f080459 app:id/view_gotacall_status_gl2jniview_localevideo}]
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	--xiaomi--lc---[M:setNativePreviewWindowId]: window:[0xa2fb5a50] start ...
	--xiaomi--lc--: setNativePreviewWindowId: video_stream_started ...
	[Legacy Capture] Preview capture window changed (oldwin: 0x1006a2 newwin: 0x2006a2 rotation:0)
	stopRecording(android.hardware.Camera@33d72037)
	[ogl_display] ogl_display_new : 0xa2beb300
	[ogl_functions] Function : Fail to load plugin libGLEW.so: dlopen failed: library "libGLEW.so" not found
	[ogl_display] init opengles_display (720 x 1280, gl initialized:0)
	[ogl_display] resize opengles_display (720 x 1280, gl initialized:0)
	[ogl_display] Falling back to legacy shaders.
	[ogl_display] OpenGL program info: 
	[TextureView Display][Filter=0xae883f58] Context initialized for windowId 0x10094a
	startRecording(0, 352, 288, 30, 0, 2726716928)
	[Capture TextureView] Rotating preview texture by 0
	[Capture TextureView] Rotating preview texture by 0
	FOCUS_MODE_CONTINUOUS_VIDEO is supported, let's use it
	Searching for closest fps range from 30000
	The closest fps range is w=7500,h=30000
	Camera preview orientation: 90
	Returning camera object: android.hardware.Camera@4a152c2
	mediastreamersetPreviewDisplaySurface(android.hardware.Camera@4a152c2, com.xiaomi.ui.call.GotaTextureView{12e5aa57 V.ED..C. ........ 507,240-705,482 #7f080459 app:id/view_gotacall_status_gl2jniview_localevideo})
	Camera [android.hardware.Camera@4a152c2] preview started
	Camera video preview size changed on videostream [0xae87d840]
	--xiaomi--lc---[M:setNativePreviewWindowId]: end ...
	MediaCodecDecoder: no buffer available for queuing this frame ! Decoder is too slow.
	MediaCodecDecoder: no buffer available for queuing this frame ! Decoder is too slow.
	MediaCodecDecoder: no buffer available for queuing this frame ! Decoder is too slow.
	Request sending of PLI on videostream [0xae87d840]
	[ogl_display] allocate_gl_textures: allocated new textures[0] (512 x 512)
	Sending RTCP SR compound message on session [0xae815600].
	jitter buffer stable: target-size: 343.366653 ms, effective-size: 103.116890 (min: 0 nom: 0, max: 500)
	jitter buffer rls stats: count=632, clockrate=90000, offset=1790.93 clock_ratio=1.01023, capped_offset=-1956534414 capped_clock_ratio=1.010226, max_ts_deviation=30903.000000 prev_max_ts_deviation=30300.000000, deviation=8614, RLS VARIABLES: P[0][0]=0.000000, P[1][0]=-0.000000, P[0][1]=-0.000000, P[1][1]=0.005948
	 H26x decoder: FPS: 8.198683
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	Captured mean fps=5.096605, expected=15.000000
	MSVideo MSTicker: We are late of 416 miliseconds.
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	H264ParameterSetsInserter: I-frame detected
	H264ParameterSetsInserter: parameter sets inserted
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	Sending RTCP SR compound message on session [0xae80f400].
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	Sending RTCP SR compound message on session [0xae815600].
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	[Legacy Capture] compute_image_rotation_correction: 90 - 0
	--xiaomi--lc--[M:unInitAudio]: start
	ms_ticker_set_time_func: ticker's time method updated.
	[Linphonexiaomi] Rendering surface texture [android.graphics.SurfaceTexture@3ff3a9a4] destroyed
	[Linphonexiaomi] Current rendering surface texture is no longer available
	[Linphonexiaomi] Releasing window surface texture [android.graphics.SurfaceTexture@3ff3a9a4]
	[Linphonexiaomi] Preview surface texture [android.graphics.SurfaceTexture@1147630d] destroyed
	[Linphonexiaomi] Current preview surface texture is no longer available
	[Linphonexiaomi] Releasing preview window surface texture [android.graphics.SurfaceTexture@1147630d]
	ms_ticker_synchronizer_destroy(): sound/wall clock skew was in average=4.050164 ms
	Filter MSRtpRecv is not scheduled; nothing to do.
	Stopping AudioStream.
	MediaStream[0xae85ef80] (MSAudio) with RtpSession[0xae80f400] summary:
	send-ssrc = [dec:820521544 hex:30e82a48]
	recv-ssrc = [dec:2545277274 hex:97b5d95a]
	===========================================================
						 RTP STATISTICS                          
	-----------------------------------------------------------
	sent                                        652 packets
												  0 duplicated packets
											  29340 bytes  
	received                                    640 packets
												  0 duplicated packets
											  28800 bytes  
	incoming delivered to the app             28800 bytes  
	incoming cumulative lost                      0 packets
	incoming received too late                    0 packets
	incoming bad formatted                        0 packets
	incoming discarded (queue overflow)           0 packets
	sent rtcp                                     5 packets
	received rtcp                                 4 packets
	===========================================================
	ms_filter_unlink: MSAndSoundRead:0xae883778,0-->MSResample:0xae8832e0,0
	ms_filter_unlink: MSResample:0xae8832e0,0-->MSAudioMixer:0xae883890,0
	ms_filter_unlink: MSAudioMixer:0xae883890,0-->MSAmrEnc:0xae883430,0
	ms_filter_unlink: MSAmrEnc:0xae883430,0-->MSRtpSend:0xae883900,0
	ms_filter_unlink: MSRtpRecv:0xae883708,0-->MSAmrDec:0xae883200,0
	ms_filter_unlink: MSAmrDec:0xae883200,0-->MSTee:0xae883548,0
	ms_filter_unlink: MSTee:0xae883548,0-->MSResample:0xae8833c0,0
	ms_filter_unlink: MSResample:0xae8833c0,0-->MSAndSoundWrite:0xae883938,0
	ms_filter_unlink: MSAudioMixer:0xae8834a0,1-->MSResample:0xae883660,0
	ms_filter_unlink: MSResample:0xae883660,0-->MSUlawEnc:0xae883698,0
	ms_filter_unlink: MSUlawEnc:0xae883698,0-->MSMKVRecorder:0xae883580,1
	ms_filter_unlink: MSItcSource:0xae8835b8,0-->MSMKVRecorder:0xae883580,0
	ms_filter_unlink: MSAudioMixer:0xae883890,1-->MSAudioMixer:0xae8834a0,0
	ms_filter_unlink: MSTee:0xae883548,1-->MSAudioMixer:0xae8834a0,1
	ms_filter_unlink: MSAudioMixer:0xae8834a0,0-->MSFileRec:0xae883468,0
	=================================================================================
							 FILTER USAGE STATISTICS                                 
	Name                          Count           Time/tick (ms)            CPU Usage
											min     mean    max     sd               
	---------------------------------------------------------------------------------
	MSAndroidVideoCapture         1303      0.00    1.22    387.79  19.80        37.6
	MSAmrEnc                      666       0.25    0.44    4.73    0.36         13.7
	MSMediaCodecH264Decoder       1303      0.00    0.41    4.82    0.67         12.7
	MSMediaCodecH264Encoder       1303      0.00    0.40    7.42    0.68         12.2
	MSAndroidTextureDisplay       238       0.00    0.31    6.26    1.03          9.5
	MSAmrDec                      297       0.07    0.24    1.73    0.18          7.4
	MSRtpSend                     2677      0.00    0.06    24.95   0.50          2.0
	MSAndSoundWrite               297       0.01    0.06    6.07    0.36          1.9
	MSResample                    1062      0.00    0.03    0.78    0.07          1.0
	MSRtpRecv                     2677      0.01    0.03    2.90    0.09          1.0
	MSUlawEnc                     99        0.00    0.01    0.04    0.01          0.2
	MSTee                         628       0.00    0.01    1.71    0.07          0.2
	MSAndSoundRead                1375      0.00    0.01    0.37    0.01          0.2
	MSMKVRecorder                 49        0.00    0.00    0.01    0.00          0.1
	MSFileRec                     99        0.00    0.00    0.01    0.00          0.1
	MSAudioMixer                  2750      0.00    0.00    0.05    0.00          0.1
	MSPixConv                     162       0.00    0.00    0.01    0.00          0.1
	MSItcSource                   1375      0.00    0.00    0.08    0.00          0.0
	MSItcSink                     0         0.00    0.00    0.00    0.00          0.0
	MSWebRTCAECM                  0         0.00    0.00    0.00    0.00          0.0
	=================================================================================
	[TextureView Display][Filter=0xae883f58] eglSwapBuffers failure for windowId 0x10094a: 0
	MSAudio MSTicker thread exiting
	MKVRecorder: destroyed
	--xiaomi--lc--[M:unInitAudio]: end
	--xiaomi--lc--[M:unInitVideo]: start ...
	[Legacy Capture] Postprocessing of Android VIDEO capture filter
	stopRecording(android.hardware.Camera@4a152c2)
	[TextureView Display][Filter=0xae883f58] eglSwapBuffers failure for windowId 0x10094a: 0
	[TextureView Display][Filter=0xae883f58] eglSwapBuffers failure for windowId 0x10094a: 0
	msasync.c: cancelling task 0xae91b6e0
	Filter MSRtpRecv is not scheduled; nothing to do.
	Stopping VideoStream
	MediaStream[0xae87d840] (MSVideo) with RtpSession[0xae815600] summary:
	send-ssrc = [dec:3149807574 hex:bbbe3fd6]
	recv-ssrc = [dec:3778639521 hex:e13976a1]
	===========================================================
						 RTP STATISTICS                          
	-----------------------------------------------------------
	sent                                        638 packets
												  0 duplicated packets
											 684333 bytes  
	received                                    775 packets
												  0 duplicated packets
											 831617 bytes  
	incoming delivered to the app            831617 bytes  
	incoming cumulative lost                      0 packets
	incoming received too late                    0 packets
	incoming bad formatted                        0 packets
	incoming discarded (queue overflow)           0 packets
	sent rtcp                                     8 packets
	received rtcp                                 4 packets
	===========================================================
	ms_filter_unlink: MSAndroidVideoCapture:0xae883af8,0-->MSPixConv:0xae883d60,0
	ms_filter_unlink: MSPixConv:0xae883d60,0-->MSTee:0xae883d28,0
	ms_filter_unlink: MSTee:0xae883d28,0-->MSMediaCodecH264Encoder:0xae883b30,0
	ms_filter_unlink: MSMediaCodecH264Encoder:0xae883b30,0-->MSRtpSend:0xae883a50,0
	ms_filter_unlink: MSRtpRecv:0xae883fc8,0-->MSTee:0xae883a88,0
	ms_filter_unlink: MSTee:0xae883a88,1-->MSItcSink:0xae883ac0,0
	ms_filter_unlink: MSTee:0xae883a88,0-->MSMediaCodecH264Decoder:0xae883e78,0
	ms_filter_unlink: MSMediaCodecH264Decoder:0xae883e78,0-->MSAndroidTextureDisplay:0xae883f58,0
	ms_filter_unlink: MSTee:0xae883d28,1-->MSAndroidTextureDisplay:0xae883f58,1
	MSVideo MSTicker thread exiting
	msasync.c: worker thread has no tasks.
	ms_yuv_buf_allocator_free(): leaving 1 mblk_t still ref'd, possible leak.
	msasync.c: waiting for task 0xae8efd60
	[TextureView Display][Filter=0xae883f58] Destroying context for windowId 0x10094a
	[ogl_display] uninit opengles_display (gl initialized:1)
	[TextureView Display][Filter=0xae883f58] OGL display destroyed
	[TextureView Display][Filter=0xae883f58] EGL display destroyed for windowId 0x10094a
	[TextureView Display][Filter=0xae883f58] Window released for windowId 0x10094a
	[TextureView Display][Filter=0xae883f58] Surface released for windowId 0x10094a
	[TextureView Display][Filter=0xae883f58] Surface destroyed for windowId 0x10094a
	[TextureView Display][Filter=0xae883f58] Context destroyed for windowId 0x10094a
	[TextureView Display][Filter=0xae883f58] Releasing global ref on windowId 0x10094a
	msasync.c: worker is signaling a thread waiting for completion.
	msasync.c: worker thread has no tasks.
	[Legacy Capture] Uninit of Android VIDEO capture filter
	--xiaomi--lc--[M:unInitVideo]: end ...
	--xiaomi--lc---[M:setNativeVideoWindowId]: window:[0x0] start ...
	--xiaomi--lc---[M:setNativeVideoWindowId]: end ...
	--xiaomi--lc---[M:setNativePreviewWindowId]: window:[0x0] start ...
	--xiaomi--lc---[M:setNativePreviewWindowId]: end ...

6. **Adb logcat logs** (mandatory if native crash)
	The APP runs normally and does not crash.

**Comment by wmailn** [comment]
I have both nativeWindow and captureWindow Windows in the android-camera2-capture.cpp file, Does this mean that separate capture and preview Windows are considered and supported?

**Comment by wmailn** [comment]
Thank you very much for your reply.
	I carefully read what you said about OutgoingCallActivity.kt file and its implementation, I guess I didn't make myself clear.
	What I want to say is whether the local capture video to the camera window and the window used to display the local video can be divided into two Windows. Instead of displaying the local video and displaying the remote video two Windows.

**Comment by wmailn** [comment]
For example, code like the following
`	int oesTextureId = GlUtil.generateTexture(GLES11Ext.GL_TEXTURE_EXTERNAL_OES);
	SurfaceTexture surfaceTexture = new SurfaceTexture(oesTextureId);
	camera.startCamera(surfaceTexture ,cameraId);`

When the camera started, it was a TEXTURE created internally instead of the actual preview window, and then passed the camera data to the actual preview window to avoid the camera opening and closing as the preview window was repeatedly created.

**Comment by Viish** [comment]
No, I don't think so.
On Android you have to use and display the preview texture created by the Android API, it's a security created to prevent a malicious app from capturing the video without the user knowing he is being filmed.

**Comment by wmailn** [comment]
> No, I don't think so. On Android you have to use and display the preview texture created by the Android API, it's a security created to prevent a malicious app from capturing the video without the user knowing he is being filmed.

I found that camera1 had this limitation, but camera2 doesn't have this limitation anymore.

**Comment by Viish** [comment]
Sorry but we won't do that.

**Comment by wmailn** [comment]
> Sorry but we won't do that.

I understand. Thank you very much.
